### PR TITLE
feat: add support for loading private and public keys

### PR DESCRIPTION
**Pull request by ajpauwels**
A default key will be either created and stored or loaded from the store on app start

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -47,6 +52,68 @@ async fn main() {
...
+            let mut sk_arr: [u8; 64] = [0; 64];
+            assert!(pk_file.read(&mut pk_arr).unwrap() == 64);
+            assert!(sk_file.read(&mut sk_arr).unwrap() == 64);
+            (pk_arr, sk_arr)
+        }
+        None => {
+            let keys = SodiumOxideKeypairGenerator::create().unwrap();
+            let pk = keys.0;
+            let sk = keys.1;
+            let new_pk_path = PathBuf::from(&public_keys_path).join(PathBuf::from("admin.pub"));
```
Comment (howhelan) [inline src/main.rs:107]: same as before, should it be `default_key_name` + ".pub" here?

File: src/crypto.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    fn create() -> Result<([u8; 64], [u8; 64]), serde_json::Error> {
+        let (pk, sk) = crypto::box_::gen_keypair();
+        // Discussion about iter->fized-size array conversion here:
+        // https://github.com/rust-lang/rust/issues/81615
+        let pk_arr: [u8; 64] = pk
+            .as_ref()
+            .iter()
+            .map(|x| *x)
+            .collect::<Vec<u8>>()
+            .try_into()
```
[the remark this answers is not in the document]
Comment (ajpauwels) [reply src/crypto.rs:31]: see comment above that block for discussion.
in short: this converts the `Vec<u8>` type into a static-sized `[u8; 64]` array type.

File: src/main.rs [hunk 3]
```diff
@@ -47,6 +52,68 @@ async fn main() {
...
+                e
+            );
+            "admin".to_owned()
+        }
+    };
+
+    let (pk, sk) = match public_keys_filtered
+        .paths
+        .iter()
+        .find(|&path| path.file_stem().unwrap() == "admin")
```
[the remark this answers is not in the document]
Comment (ajpauwels) [reply src/main.rs:88]: yes

File: src/main.rs (hunk 1 above)
Comment (ajpauwels) [reply src/main.rs:107]: yes

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ajpauwels**
You can assign the PR to them by writing `/assign @ajpauwels` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=pauwels-labs%2Fredact-client).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-client/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ajpauwels](https://github.com/pauwels-labs/redact-client/pull/15#issuecomment-832590323 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=pauwels-labs%2Fredact-client).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/pauwels-labs/redact-client/blob/master/OWNERS)~~ [ajpauwels]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**